xform-anatomy.rst

```diff
@@ -0,0 +1,420 @@
+
+.. code-block:: xml
+
+  <data id="myform" >
+    <meta>
+      <instanceID>uuid:23b56e39-ef50-4510-b85f-c454cd5465c1</instanceID>
+    </meta>
+    <mystring>Hello World!  This is my answer!</mystring>
+  </data>
+
+Beginning with ODK Collect 1.2, there can be multiple instances defined in the form. The second and subsequent instance definitions can specify static data used in filling out the form (e.g., lists of cities within each county within a given state, dosage tables). The most common use would be to present the choices in a cascading select, e.g., where you are asked to first choose a state, then the county within that state, then the city within that county. 
+
+If you specify a value for :``<mystring>`` in the original XForm, it presents that as the default answer to the prompt:
+
+.. code-block:: xml
+
+  <mystring>Default answer</mystring>
+
+Adding more prompts is as simple as adding more variables:
+
+.. code-block:: xml
+
+  <instance>
+  <data id="myotherform" >
+    <meta>
+      <instanceID/>
+    </meta>
+    <mystring/>
+    <a_number/>
+    <birthday/>
+    <date>2010-06-15</date>
+    <select>a c</select>
+    <favorite_number>7</favorite_number>
+    <whatever_i_want/>
+  </data>
+  </instance>
+
+.. form-bindings:
+
+Bindings
+---------
+
+Since we have two variables, we need two bindings:
+
+.. code-block:: html
+
+  <bind nodeset="/data/meta/instanceID" type="string" 
+           readonly="true()" calculate="concat('uuid:',uuid())" />
+  <bind nodeset="/data/mystring" type="string"/>
+
+The first binding defines how the unique identifier for a filled-in form will be constructed.
+
+The second binding specifies that ``<mystring>`` will hold a string value. Bindings that do not specify a "type" are assumed to be strings. However, ODK Aggregate issues warnings if a type is not specified because ordering of numbers differs from ordering of strings; specifying a "type" ensures that the proper ordering is applied, i.e., "10" is ordered alphabetically less than "2", but, if these are interpreted as numbers, the order is reversed (10 is greater than 2).
+
+Bindings are also where you specify branching, calculations, read-only fields, required fields, input constraints, etc.
+
+Note that the binding references the instance node using an XPath expression. The expression represents the hierarchy inside the ``<instance>`` tags.
+
+.. code-block:: html
+
+  <instance>
+  <data>
+    <mystring/> <!-- referenced as /data/mystring -->
+  </data>
+  </instance>
+
+In this example, mystring is referenced by: :guilabel:`/data/mystring`.
+
+.. form-body:
+
+Body
+-----
+
+The body represents what should be shown to the user. In this example, we have two variables, but the ``<instanceID>`` variable is merely for bookkeeping. Thus, we will only have one prompt displayed to the user for the ``<mystring>`` variable. This is placed in the body element:
```

Comment:
Not variables.

We have two pieces of data in the form, but `instanceID` is not editable by the user, so it will not appear in the rendered form.

Reply:
is there a specific name for this "piece of data"? I know it's not a `field`.